app/src/main/java/com/example/bookwormadventuresdeluxe2/AddOrEditBooksActivity.java

```diff
@@ -36,6 +38,26 @@ protected void onCreate(Bundle savedInstanceState)
         authorView = findViewById(R.id.author_edit_text);
         descriptionView = findViewById(R.id.description_edit_text);
         isbnView = findViewById(R.id.isbn_edit_text);
+
+        // If editing a book, prepopulate text fields with their old values
+        int requestCode = -1;
+        if (getIntent().getSerializableExtra("requestCode") != null)
```

Comment:
For using integers with intent, I think best practice is to use [putExtra()](https://stackoverflow.com/questions/7074097/how-to-pass-integer-from-one-activity-to-another) in the parent activity and using getIntExtra() for the child activity. Using getSerializable(...) is usually used for passing in user defined objects around activities.

Reply:
Nice catch, fixed